lib/protractor.js

```diff
@@ -661,34 +661,30 @@ Protractor.prototype.initDebugger_ = function(debuggerClientPath, opt_debugPort)
     return asString;
   };
 
-  if (opt_debugPort) {
-    process.debugPort = opt_debugPort;
-  }
-
-  // Call this private function instead of sending SIGUSR1 because Windows.
-  process._debugProcess(process.pid);
-
-  var flow = webdriver.promise.controlFlow();
-  var self = this;
-  var pausePromise = flow.execute(function() {
-    log.puts('Starting WebDriver debugger in a child process. Pause is ' +
-        'still beta, please report issues at github.com/angular/protractor\n');
-    var args = [process.debugPort];
-    if (self.debuggerServerPort_) {
-      args.push(self.debuggerServerPort_);
-    }
-    var nodedebug = require('child_process').fork(debuggerClientPath, args);
-    process.on('exit', function() {
-      nodedebug.kill('SIGTERM');
+  // Invoke fn if port is available.
+  var onPortAvailable = function(port, fn) {
+    var net = require('net');
+    var tester = net.connect({port: port}, function() {
```

Comment:
Ah sorry, I meant something like this: https://github.com/SeleniumHQ/selenium/blob/master/javascript/node/selenium-webdriver/net/portprober.js#L159

Except we don't actually want to create a server, so the method you had is the best I can think of.